scene/resources/surface_tool.cpp

```diff
@@ -543,7 +543,7 @@ Array SurfaceTool::commit_to_arrays() {
 				for (uint32_t idx = 0; idx < vertex_array.size(); idx++) {
 					const Vertex &v = vertex_array[idx];
 
-					if (v.bones.size() > count) {
+					if (v.bones.size() < count) {
```

Comment:
Excluding `v.bones.size() < count` is correct to prevent out of read exception indeed. The main question is whether `v.bones.size() > count` is de facto a valid case, or should also be excluded (thus with `!=`). CC @fire

Reply:
A fast check of the [documentation regarding the ArrayType enum](https://docs.godotengine.org/en/stable/classes/class_arraymesh.html#class-arraymesh-constant-array-max), located inside the ArrayMesh class, insinuates that the correct way to proceed is by treating inequality as invalid.

Reply:
When the bone_size 4 is less than count of 8, trigger printing "Invalid bones size %d vs count %d", v.bones.size(), count)) and continue.

I think this is a logic error, and will cause all bones to not have bone weights.

Can you check?

Reply:
I cannot check it, but from the code it seems to be the case. 
The way to take care of it would be just by assigning the most of the heights, and proceed to skip the excess, or somehow repeat weights for the excess bones.